tests/test_deltacode.py

```diff
@@ -1615,4 +1637,28 @@ def test_Stat_calculation_and_ordering(self):
 
         stats_object = deltacode_object.stats.to_dict()
 
-        assert stats_object == expected
\ No newline at end of file
+        assert stats_object == expected
+
+    def test_similarity_matching_1(self):
+        old_file = self.get_test_loc('deltacode/coala-0.7.0-old.json')
+        new_file = self.get_test_loc('deltacode/coala-0.10.0-new.json')
+        result_file = self.get_temp_file('json')
+        args = ['--new', old_file, '--old', new_file, '--json-file', result_file, '--all-delta-types']
+        cli_test_utils.run_scan_click(args)
+        cli_test_utils.check_json_scan(self.get_test_loc('deltacode/coala-expected-result.json'), result_file, regen=False)
+
+    def test_similarity_matching_2(self):
+        old_file = self.get_test_loc('deltacode/sugar-0.108.0-old.json')
+        new_file = self.get_test_loc('deltacode/sugar-0.114-new.json')
+        result_file = self.get_temp_file('json')
+        args = ['--new', old_file, '--old', new_file, '--json-file', result_file, '--all-delta-types']
+        cli_test_utils.run_scan_click(args)
+        cli_test_utils.check_json_scan(self.get_test_loc('deltacode/sugar-expected.json'), result_file, regen=False)
+
+    def test_non_similarity_matching_1(self):
+        old_file = self.get_test_loc('deltacode/coala-0.7.0-old.json')
+        new_file = self.get_test_loc('deltacode/sugar-0.114-new.json')
+        result_file = self.get_temp_file('json')
+        args = ['--new', old_file, '--old', new_file, '--json-file', result_file, '--all-delta-types']
```

Comment:
Why do we make `old_file` the argument to `--new` and vice-versa for the `--old` argument? We should keep these names aligned (i.e. `args = ['--new', new_file, '--old', old_file, '--json-file', result_file, '--all-delta-types']`